src/hardware/adc.rs

```diff
@@ -84,6 +84,28 @@ use hal::dma::{
     MemoryToPeripheral, PeripheralToMemory, Transfer,
 };
 
+/// A type representing an ADC sample.
+#[derive(Copy, Clone)]
+pub struct AdcSample(pub u16);
```

Comment:
Why `Sample` for one but `Code` for the other? Please use `Code` for both.
And `impl From<AdcCode> for f32` instead of `Into` as suggested.

Reply:
Or doesn't that work because of the orphaning rules?

Reply:
Given that the IIR coefficients handle the mapping of f32 -> code, I assumed we could forgo explicitly providing that type conversion.

Reply:
DAC output data are samples with the exactly the same semantics as ADC input data samples.
Ack the `Into` preference!
To have SI semantics for the IIR filter we need to convert `y_offset`, `y_min` and `y_max`. The coefficients are agnostic to scaling because the scaling is the same here (apart from the AFE gain which I don't want to bring into the IIR equation).

Reply:
Let's defer the IIR parameter refactoring to another time.

Reply:
Yeah, I'm still not convinced the IIR needs to change anyways, since the coefficients are pretty much meaningless numbers to a human anyways and are generally calculated by some other utility.